### DISCUSSION TITLE
There could be a fingerprnt on start to see if something on device or microSD changed

### DISCUSSION THREAD

**Discussion by DiosDelRayo** [XmrSigner/xmrsigner#13] [Ideas]
This feature would take a sha256(password + serial number + microSD content) and from there the last 6 digits

**Comment by kevin0t** [comment]
how secure and trustable would this process be without the use of something like hardware based secure element ex: a tpm chip which supports verified boot ? If an attacker gains access to the microsd card they could read the fingerprint and then modify the os to lie about serial number or microsd content while the os being malicios. 

The OS fingerprint usually is verified against something like a secure element and/or tpm. Maybe i missed something in this idea

**Reply by DiosDelRayo** [reply]
How the attacker should not know the password, how he would prepare the modified microSD card content in a way that the sha256 is still the same (with the same password)? Why would you trust any hardware more then yourself?

The idea is that you have your device (you want to be sure to verify that it was not exchanged, not 100% sure with this method, but I think as far and easy it will get) and that the microSD was not tempered with. So as attacker you can only show the right 6 digits if you have the password, the serial number and the unmodified microSD, as long one is missing, I see no way how to show the user the right 6 digits on tempering the microSD card or exchanging the pi.
